tests/unit/nupic/algorithms/knn_classifier_test.py

```diff
@@ -29,6 +29,34 @@
 
 class KNNClassifierTest(unittest.TestCase):
 
+ 
+  
+  
+  
+  def testSparsifyVector(self):
```

Comment:
This test appears to have four separate tests. I would separate them into different test methods. As it is, if you had a typo when redefining `classifier` it would be very hard to discover since it would just use the one defined by the previous code block. Separating into separate methods ensures there aren't any weird interactions like this.